example/src/main/resources/templates/webeid-callback.html

```diff
@@ -0,0 +1,103 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8"/>
+    <meta name="viewport" content="width=device-width, initial-scale=1"/>
+    <meta http-equiv="Cache-Control" content="no-store"/>
+    <meta id="csrftoken" name="csrftoken" th:content="${_csrf.token}"/>
+    <meta id="csrfheadername" name="csrfheadername" th:content="${_csrf.headerName}"/>
+    <title>Completing signing…</title>
+    <link rel="stylesheet" href="/css/bootstrap.min.css"/>
+    <link rel="stylesheet" href="/css/main.css"/>
+</head>
+
+<body class="loading-page">
+<div id="spinner-message">
+    <h2>Completing signing…</h2>
+    <div class="spinner"></div>
+</div>
+<div id="error-message" class="alert alert-danger" style="display: none;" role="alert">
+    <div class="message"></div>
+    <pre class="details"></pre>
+</div>
+
+<p class="text-center p-4" style="display: none;" id="error-actions">
+    <button id="back-button" class="btn btn-primary">Back</button>
+</p>
+
+<script type="module">
+    import {showErrorMessage, checkHttpError} from '/js/errors.js';
+
+    // Using an async IIFE for mobile WebView compatibility:
+    // top-level await is not supported in some mobile browsers/WebViews.
+    (async () => {
+        const fragment = window.location.hash.slice(1);
+
+        if (!fragment) {
+            throw new Error("Missing response payload");
+        }
+
+        let payload;
+        try {
+            payload = JSON.parse(atob(fragment));
+        } catch (e) {
+            console.error(e)
+            throw new Error("Failed to parse the response");
+        }
+
+        if (payload.error) {
+            const error = new Error(payload.message ?? "Signing failed");
+            error.code = payload.code;
+            throw error;
+        }
```

Comment:
Let's make code DRYer by extracting payload parsing into a separate function and reusing it here and in `webeid-login.html`:

```suggestion
        const payload = parsePayload("Signing");
```